account_banking_sepa_credit_transfer/models/account_payment_order.py

```diff
@@ -191,10 +191,11 @@ def generate_payment_file(self):  # noqa: C901
                     gen_args,
                     line,
                 )
-                line_purpose = line.payment_line_ids[:1].purpose
-                if line_purpose:
-                    purpose = etree.SubElement(credit_transfer_transaction_info, "Purp")
-                    etree.SubElement(purpose, "Cd").text = line_purpose
+                payment_line = line.payment_line_ids[0]
```

Comment:
Why changing this? Are you sure there's always going to be a payment_line_id?

Reply:
It's just the same code that was moved from account_banking_sepa_credit_transfer to account_banking_pain_base, to mutualize the code with account_banking_sepa_direct_debit. On a payment order, an account.payment is always linked to N account.payment.line, and we read the purpose on the account.payment.line because that's where the information is. I didn't change the logic.